cloudinit/cmd/main.py

```diff
@@ -276,36 +305,6 @@ def attempt_cmdline_url(path, network=True, cmdline=None) -> Tuple[int, str]:
     )
 
 
-def purge_cache_on_python_version_change(init):
```

Comment:
This makes more sense as an Init() method. A few other functions in this file should probably be moved as well (`_should_bring_up_interfaces()`, `_maybe_persist_instance_data()`, `_maybe_set_hostname()`), but this PR is already too big.